### DISCUSSION TITLE
Branches

### DISCUSSION THREAD

**Discussion by icarusinthewings** [CatalystDevOrg/Catalyst#43] [Proposals]
Idea: Add separate branches for commits to go through
This would create a stable workflow. Commits would start in the "dev" branch, be merged into the "beta" branch when they're stable enough, and then be merged into the "main" branch when they're ready to be live and are well-tested. Builds of the "dev" branch could occur daily (by implementing Github Actions), "beta" could occur weekly, and "main" could occur whenever a hotfix is ready or whenever it's time for a new build.

**Comment by jdev082** [comment]
dev and beta channels of software updates are both distributed as builds, and packaged into releases. From their, the source code of the release can be downloaded. Also an option, going into the branches interface and clicking the tags tab will allow you to view the source code of the release.

Your method does not seem necessary.

**Reply by hello-smile6** [reply]
> Your method does not seem necessary.

It works for Chromium, it could work for this.

**Reply by jdev082** [reply]
@hello-smile6 a system is already in place, a system I prefer.

**Reply by imdevarsh** [reply]
I like the current system.

**Comment by jdev082** [comment]
This is being tested with v3.10s development cycle.
